### PR TITLE
Next review iteration

### PR DESCRIPTION
**Pull request by storojs72**
This PR introduces next review iteration of Acra docs from my side. 

Apart from traditional fixes of poor phrases, grammar and typos I also rephrased a bit data re-encryption subsection and moved it to more appropriate place, as we discussed in Slack channel.

### CODE REVIEW

File: acra/acra-in-depth/architecture/key-storage-and-kms/key-storage.md [hunk 1]
```diff
@@ -51,4 +51,4 @@ We recommend deploying a dedicated Redis database as key storage and token stora
 
 When AcraServer/AcraTranslator/AnyProxy need a key, they ask key storage for it. Each key has its purpose (transport, encryption, audit log, HMAC) and is bound to a specific client ID or zone. Read more about [key management](/acra/security-controls/key-management/).
 
-AcraServer and AcraTranslator will require key storage for tokenization. When [tokenization](/acra/security-controls/tokenization/) is performed, the key storage will contain data needed for consistent tokenization (return same result for same input next time) and for detokenization (reverse operation).
+AcraServer and AcraTranslator will require key storage for tokenization. When [tokenization](/acra/security-controls/tokenization/) is performed, the key storage will contain data needed for consistent tokenization (return same result for same input next time) and for de-tokenization (reverse operation).
```
Comment (vixentael) [inline acra/acra-in-depth/architecture/key-storage-and-kms/key-storage.md:54]: we use `detokenization` term everywhere, both are ok, `detokenization` is more popular and it's across all the docs and website

Screenshot 2021-11-29 at 21 33 45

Comment (vixentael) [inline acra/acra-in-depth/architecture/key-storage-and-kms/key-storage.md:54]:
```suggestion
AcraServer and AcraTranslator will require key storage for tokenization. When [tokenization](/acra/security-controls/tokenization/) is performed, the key storage will contain data needed for consistent tokenization (return same result for same input next time) and for detokenization (reverse operation).
```

File: acra/security-controls/_index.md [hunk 2]
```diff
@@ -8,19 +8,19 @@ weight: 7
...
 
-This section describes how Acra's features and working and how to use them. Select a feature depending on your use case.
+This section describes how Acra's features are working and how to use them properly. Select a feature depending on your use case.
 
 ## Protect the data
 
 Encrypt data before putting it into your database/datastore.
 
-* [Encryption](/acra/security-controls/encryption/). Use encryption if your goal is to protect sensitive data fields. Acra will encrypt data fields and allow you to store them encrypted as binary blobs in your database/datastore. This is the main feature most Acra users use.
+* [Encryption](/acra/security-controls/encryption/). Use encryption if your goal is to protect confidentiality of sensitive data fields. Acra will encrypt data fields and allow you to store them encrypted as binary blobs in your database/datastore. This is the main feature most Acra users use.
```
Comment (vixentael) [inline acra/security-controls/_index.md:17]:
```suggestion
* [Encryption](/acra/security-controls/encryption/). Use encryption if your goal is to protect confidentiality and integrity of sensitive data fields. Acra will encrypt data fields and allow you to store them encrypted as binary blobs in your database/datastore. This is the main feature most Acra users use.
```

We use AES-GCM

File: acra/security-controls/_index.md [hunk 3]
```diff
@@ -8,19 +8,19 @@ weight: 7
...
+* [Masking](/acra/security-controls/masking/). Use masking if your goal is to encrypt and hide data, but leave a part of it in a plaintext (`annXXXXX@acme.com`). Your application controls how to show data – in plaintext for trusted connections, or as a mask for untrusted ones.
 
-* [Tokenization](/acra/security-controls/tokenization/). Use tokenization for data fields when you can't change data column format. Acra will create a data token with the same format (int, string, or even email-looking) and store it in the database. Acra will encrypt the data fields and store them in separate storage.
+* [Tokenization](/acra/security-controls/tokenization/). Use tokenization for data fields when you want a ciphertext to have the same format as plaintext. Acra will create a data token with the same format (int, string, or even email-looking) and store it in the database. Acra will encrypt the data fields and store them in separate storage.
```
Comment (vixentael) [inline acra/security-controls/_index.md:23]: Not really. The ciphertext format is binary, it's stored outside the main DB, in Redis. Main DB has tokens, which are stored in the same format as original data.

```suggestion
* [Tokenization](/acra/security-controls/tokenization/). Use tokenization for data fields when you can't change data column format. Acra will create a data token with the same format (int, string, or even email-looking) and store it in the database. Acra will encrypt the data fields and store them in separate storage.
```

File: acra/security-controls/encryption/_index.md [hunk 4]
```diff
@@ -5,28 +5,28 @@ weight: 1
 
 # Application-level encryption
 
-Acra prevents adversaries with access to database from seeing sensitive plaintexts, if it has been configured so.
-Acra does that by encrypting the sensitive data selectively and keeping the keys to itself. In this, Acra acts as security boundary in front of sensitive data. 
+Acra can be configured to prevent adversaries with access to database from seeing sensitive plaintexts.
```
Comment (vixentael) [inline acra/security-controls/encryption/_index.md:8]:
```suggestion
Acra prevents adversaries with access to database from seeing sensitive data in plaintext.
```

let's be more assertive in our docs

File: acra/security-controls/encryption/_index.md [hunk 5]
```diff
@@ -5,28 +5,28 @@ weight: 1
...
+Acra selectively encrypts sensitive data using securely stored cryptographic keys. Thus, Acra acts as a security boundary in front of sensitive data.
 
-What Acra does can be described as [application-level encryption](https://www.infoq.com/articles/ale-software-architects/) on field level, or, field-level encryption.
+Essentially, Acra provides [application-level encryption](https://www.infoq.com/articles/ale-software-architects/) by implementing it as a field-level database encryption.
```
Comment (vixentael) [inline acra/security-controls/encryption/_index.md:11]:
```suggestion
Essentially, Acra provides [application-level encryption](https://www.infoq.com/articles/ale-software-architects/) by implementing it as a field-level encryption before data gets into the database.
```

"field-level encryption" is a search term, let's not ruin it.

**Review by vixentael — CHANGES_REQUESTED** [review]

File: acra/security-controls/encryption/_index.md [hunk 6]
```diff
@@ -5,28 +5,28 @@ weight: 1
...
-* AcraServer transparent encryption: if AcraServer has been set up in transparent mode, encryption happens on AcraServer during INSERT/UPDATE queries. You configure which fields should be encrypted in `encryptor_config`.
-* AcraTranslator: encryption happens as result of gRPC or HTTP API call from client application.
+* AcraServer transparent encryption: if AcraServer has been set up in a transparent mode, encryption happens on AcraServer during INSERT/UPDATE queries. You configure which fields should be encrypted in `encryptor_config`.
+* AcraTranslator: encryption happens as a result of gRPC or HTTP API call from client application.
 * [Client-side encryption using AcraWriter SDK](/acra/acra-in-depth/architecture/sdks/acrawriter/): it is possible to generate AcraStructs/AcraBlocks using AcraWriter library inside your application using appropriate keys.
 
 ## Decryption
 
-There are three ways to encrypt data from AcraStructs/AcraBlocks back in plaintext:
+There are three ways to decrypt data from AcraStructs/AcraBlocks back into a plaintext:
```
Comment (vixentael) [inline acra/security-controls/encryption/_index.md:26]: good catch!

File: acra/security-controls/encryption/_index.md [hunk 7]
```diff
@@ -43,7 +43,7 @@ When a column is configured to be encrypted, AcraServer will transparently encry
 You can also encrypt the data manually on the client side using
 [AcraWriter SDK](/acra/acra-in-depth/architecture/sdks/acrawriter/).
 
-Decryption on the other hand is done automagically: AcraServer will detect encrypted data and will attempt
+Decryption on the other hand is done automatically: AcraServer will detect encrypted data and will attempt
```
Comment (vixentael) [inline acra/security-controls/encryption/_index.md:46]: that was a joke :)

File: acra/security-controls/key-management/inventory.md [hunk 8]
```diff
@@ -31,29 +29,29 @@ Read about [key generation](/acra/security-controls/key-management/operations/ge
 ## Acra Master Key
 
 {{< hint warning >}}
-Do not lose Acra Master Key!
+Keep an eye on Acra Master Key!
```
Comment (vixentael) [inline acra/security-controls/key-management/inventory.md:32]: I want to be really explicit here..

```suggestion
Do not lose Acra Master Key!
```

File: acra/security-controls/key-management/inventory.md [hunk 9]
```diff
@@ -62,25 +60,25 @@ After reading it, Acra will store it in memory, although it can be purged and re
 ## Intermediate keys
 
 {{< hint warning >}}
-Do not lose intermediate keys.
+Keep an eye on intermediate keys!
```
Comment (vixentael) [inline acra/security-controls/key-management/inventory.md:63]: Same, it's important emphasize.

```suggestion
Do not lose intermediate keys!
```

File: acra/security-controls/key-management/inventory.md [hunk 10]
```diff
@@ -143,7 +140,7 @@ When [searchable encryption](/acra/security-controls/searchable-encryption/) is
 
 AcraServer, AcraConnector and AcraTranslator use symmetric keys to generate tamper-proof [audit log](/acra/security-controls/security-logging-and-events/audit-logging). Then [acra-verifier](/acra/security-controls/security-logging-and-events/audit-logging/) utility uses same key to verify signed logs.
 
-Under the hood and HMAC calculations to sign/verify signature of log messages.
+Under the hood this key is used for multiply chained HMAC-calculations to sign/verify log-stream generated by Acra's service.
```
Comment (vixentael) [inline acra/security-controls/key-management/inventory.md:143]:
```suggestion
Under the hood this key is used for multiply chained HMAC-calculations to sign/verify log-stream generated by Acra's services.
```

File: acra/security-controls/key-management/operations/rotation.md [hunk 11]
```diff
@@ -110,6 +110,17 @@ There're different strategies of re-encrypting data after rotating **data storag
     Re-encrypting data "piece by piece", leaving old keys active for some amount of time.
     This strategy is the best one performance-wise, but requires careful planning.
 
+Note that there are some differences in data re-encryption flow depending on selected cryptographic container type.
+**AcraStructs** use ephemeral asymmetric keys to enable applications to encrypt the data but never decrypt it (end-to-end encryption) - the private key
```
Comment (vixentael) [inline acra/security-controls/key-management/operations/rotation.md:114]:
```suggestion
[**AcraStructs**](/acra/acra-in-depth/data-structures/acrastruct/) use ephemeral asymmetric keys to enable applications to encrypt the data but never decrypt it (end-to-end encryption) - the private key
```

File: acra/security-controls/key-management/operations/rotation.md [hunk 12]
```diff
@@ -110,6 +110,17 @@ There're different strategies of re-encrypting data after rotating **data storag
...
+stays on the AcraServer / AcraTranslator which are the only ones capable of decryption. In case of AcraStruct container you
+can be sure that: 1) even if application used outdated (rotated) key, the data stored in the database is encrypted with a fresh key;
+2) it is impossible to find any correlation between encrypted data and any table row, as the re-encrypted data will look completely different.
+
+**AcraBlocks** use symmetric keys stored on AcraServer / AcraTranslator for both encryption and decryption. AcraBlocks provide similar properties for encrypted data
```
Comment (vixentael) [inline acra/security-controls/key-management/operations/rotation.md:119]:
```suggestion
[**AcraBlocks**](/acra/acra-in-depth/data-structures/acrablock/) use symmetric keys stored on AcraServer / AcraTranslator for both encryption and decryption. AcraBlocks provide similar properties for encrypted data
```

File: acra/security-controls/masking/_index.md [hunk 13]
```diff
@@ -16,13 +16,10 @@ Use cases may include:
...
-Under the hood, AcraServer/AcraTranslator encrypt the data fully, and concatenate a piece of plaintext (defined by `plaintext_length` variable, see below). So the data itself is stored encrypted into AcraStruct/AcraBlock with a tiny chunk of plaintext data.
-
-One components can provide masking functionality:
-
-* AcraServer — transparent masking for `INSERT` and `UPDATE` queries,
-  transparent demasking for `SELECT` queries, with per column configuration.
+Under the hood, AcraServer/AcraTranslator encrypts the data completely, and concatenates a piece of plaintext (defined by `plaintext_length` variable, see below). So the data itself is stored encrypted into AcraStruct/AcraBlock with a tiny chunk of plaintext data.
 
+AcraServer is the only component that provides transparent masking for `INSERT` and `UPDATE` queries and
+transparent de-masking for `SELECT` queries, with per column configuration.
```
Comment (vixentael) [inline acra/security-controls/masking/_index.md:22]:
```suggestion
transparent demasking for `SELECT` queries, with per column configuration.
```

File: acra/security-controls/tokenization/_index.md [hunk 14]
```diff
@@ -22,22 +22,22 @@ The tokenization process has few interesting properties:
...
+  * Consistent tokenization — result will always remain the same for the same input (`hello` might be tokenized into `wshfwSjdsn`),
   * Inconsistent tokenization — result will be different every time, even for the same input (`hello` might be tokenized into `KishJs` or `KdCbshQoP` or `KeitAyheof`),
 * Tokenization can be reversed, but only for valid (previously returned) values.
 
 Two components can provide tokenization functionality:
 
 * AcraServer — transparent tokenization for `INSERT` and `UPDATE` queries,
-  transparent detokenization for `SELECT` queries, with per column configuration.
-* AcraTranslator — provides gRPC and HTTP API.
+  transparent de-tokenization for `SELECT` queries, with per column configuration.
```
Comment (vixentael) [inline acra/security-controls/tokenization/_index.md:32]:
```suggestion
  transparent detokenization for `SELECT` queries, with per column configuration.
```

File: acra/security-controls/tokenization/_index.md [hunk 15]
```diff
@@ -73,7 +73,7 @@ In order to use the translator gRPC API you have to take
 [api.proto](https://github.com/cossacklabs/acra/blob/master/cmd/acra-translator/grpc_api/api.proto)
 and use either `service Tokenizator` or `service BulkProcessing` (with one or more `TokenizeRequest` inside, enterprise only).
 
-Tokenize & detokenize with `service Tokenizator`:
+Tokenize & de-tokenize with `service Tokenizator`:
```
Comment (vixentael) [inline acra/security-controls/tokenization/_index.md:76]:
```suggestion
Tokenize & detokenize with `service Tokenizator`:
```

File: acra/security-controls/tokenization/_index.md [hunk 16]
```diff
@@ -104,7 +104,7 @@ service Tokenizator {
 ```
 
 {{< hint info >}}
-Detokenization is performed by creating `TokenizeRequest` with the token as value and passing it to `Detokenize` method. 
+De-tokenization is performed by creating `TokenizeRequest` with the token as value and passing it to `Detokenize` method.
```
Comment (vixentael) [inline acra/security-controls/tokenization/_index.md:107]:
```suggestion
Detokenization is performed by creating `TokenizeRequest` with the token as value and passing it to `Detokenize` method.
```

File: acra/security-controls/tokenization/_index.md [hunk 17]
```diff
@@ -114,7 +114,7 @@ Don't forget to use the same type (str / email / int32 / int64 / bytes).
 
 Method: `GET`
 
-Path: `/v2/tokenize` for tokenization, `/v2/detokenize` for detokenization
+Path: `/v2/tokenize` for tokenization, `/v2/detokenize` for de-tokenization
```
Comment (vixentael) [inline acra/security-controls/tokenization/_index.md:117]:
```suggestion
Path: `/v2/tokenize` for tokenization, `/v2/detokenize` for detokenization
```

**Review by vixentael — CHANGES_REQUESTED** [review]
Great work! Please see my corrections

File: acra/security-controls/encryption/_index.md [hunk 18]
```diff
@@ -43,7 +43,7 @@ When a column is configured to be encrypted, AcraServer will transparently encry
 You can also encrypt the data manually on the client side using
 [AcraWriter SDK](/acra/acra-in-depth/architecture/sdks/acrawriter/).
 
-Decryption on the other hand is done automagically: AcraServer will detect encrypted data and will attempt
```
[the remark this answers is not in the document]
Comment (vixentael) [reply acra/security-controls/encryption/_index.md:46]: that was a joke 😿

File: acra/acra-in-depth/architecture/key-storage-and-kms/key-storage.md (hunk 1 above)
Comment (storojs72) [reply acra/acra-in-depth/architecture/key-storage-and-kms/key-storage.md]: Agree with such argument

**Comment by storojs72** [comment]
@vixentael thanks for your assistance! I applied all your suggestions and returned back "automagically" word

**Review by vixentael — APPROVED** [review]
awesome, lets' merge! 🚀

**Pull request merged**